packages/react-core/src/demos/ComposableMenu.md

```diff
@@ -869,14 +868,240 @@ MenuWithFlyout = () => {
+            children: group.props.children.filter(child => {
+              if (filteredIds.includes(child.props.itemId) || child.type === Divider) {
+                return child;
+              }
+            })
+          });
+          if (filteredGroup.props.children.length > 0 && filteredGroup.props.children[0].type !== Divider) {
+            return filteredGroup;
+          }
+        } else {
+          if (filteredIds.includes(group.props.itemId)) {
+            return group;
+          }
+        }
+      })
+      .filter(newGroup => newGroup);
+
+    if (filteredCopy.length > 0) {
+      let lastGroup = filteredCopy.pop();
+      if (lastGroup.type === MenuGroup) {
+        lastGroup = React.cloneElement(lastGroup, {
+          children: lastGroup.props.children.filter(item => item.type !== Divider)
+        });
+      }
+      filteredCopy.push(lastGroup);
+    }
+    return filteredCopy;
+  };
+
+  const onTextChange = textValue => {
+    if (textValue === '') {
+      setFilteredIds(['*']);
+      return;
+    }
+
+    const filteredIds = refFullOptions
+      .filter(item => !textValue || item.innerText.toLowerCase().includes(textValue.toString().toLowerCase()))
+      .map(item => item.id);
+    setFilteredIds(filteredIds);
+  };
+
+  const onFavorite = (event, itemId, actionId) => {
+    event.stopPropagation();
+    if (actionId === 'fav') {
+      const isFavorite = favorites.includes(itemId);
+      if (isFavorite) {
+        setFavorites(favorites.filter(fav => fav !== itemId));
+      } else {
+        setFavorites([...favorites, itemId]);
+      }
+    }
+  };
+
+  const filteredFavorites = filterItems(createFavorites(favorites), filteredIds);
+  const filteredItems = filterItems(menuItems, filteredIds);
+  if (filteredItems.length === 0) {
+    filteredItems.push(<MenuItem key="no-items">No results found</MenuItem>);
+  }
+
+  const menu = (
+    <Menu ref={menuRef} onActionClick={onFavorite} onSelect={(_ev, itemId) => console.log('selected', itemId)}>
+      <MenuContent>
+        <MenuList>
+          <MenuInput>
+            <TextInput
+              aria-label="Filter menu items"
+              iconVariant="search"
+              type="search"
+              onChange={value => onTextChange(value)}
+            />
+          </MenuInput>
+          <Divider component="li" />
+          {filteredFavorites.length > 0 && (
+            <MenuGroup key="favorites-group" label="Favorites">
+              {filteredFavorites}
+              <Divider key="favorites-divider" component="li" />
+            </MenuGroup>
+          )}
+          {filteredItems}
+        </MenuList>
+      </MenuContent>
+    </Menu>
+  );
   return (
-    <Popper
-      trigger={toggle}
-      popper={menu}
-      isVisible={isOpen}
-      popperMatchesTriggerWidth={false}
-    />
+    <Popper trigger={toggle} popper={menu} isVisible={isOpen} popperMatchesTriggerWidth={false} enableFlip={false} />
```

Comment:
Why no `enableFlip`?

Reply:
Ah that was just me forcing the menu to expand downwards (since its last on the page). I can re-enable it.